### PR TITLE
BS-2822: Implement Symfony middleware

### PR DESCRIPTION
**Pull request by yashtewari**

### CODE REVIEW

**Comment by yashtewari** [comment]
Hey @amirbenun, please have a look!
Note: all files except `src/OpenPolicyAgent/Authorizer.php ` are going to be purged.

**Review by amirbenun — APPROVED** [review]

File: src/OpenPolicyAgent.php [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+        // Check if the received controller method has the Authorize attribute.
+        $attribute = $method->getAttributes(
+            Authorize::class,
+            \ReflectionAttribute::IS_INSTANCEOF,
+        )[0] ?? null;
+
+        if ($attribute == null) {
+            return;
+        }
```
Comment (amirbenun) [inline src/OpenPolicyAgent.php:108]: We should think of a convention, what do we do if the expected annotation is missing?
1. We dont send the request
2. We send the request leaving that field empty

File: src/OpenPolicyAgent.php [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        $response = $this->client->request(
+            'POST',
+            $this->pdp_config['hostname'].':'.$this->pdp_config['port'].'/v1/data'.$this->pdp_config['policy.path'],
+            array(
+                'json' => $payload,
+                'timeout' => $this->pdp_config['connectionTimeout.milliseconds'],
+            ),
+        );
+
+        return ($response->toArray()['result']['allow'] == true);
```
Comment (amirbenun) [inline src/OpenPolicyAgent.php:143]: Another convention we need to figure out is if we expect the result to have `allow: true` or just `true`?

File: src/OpenPolicyAgent.php [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        }
+    }
+
+    // authorize makes the call to the PDP.
+    private function authorize($event, $resources): bool {
+        $request = $event->getRequest();
+
+        $payload = array(
+            'input' => array(
+                'request' => array(
```
Comment (amirbenun) [inline src/OpenPolicyAgent.php:121]: consider adding `source` and `destination`

**Pull request merged**